utils/clflags.ml

```diff
@@ -494,9 +494,9 @@ let print_arguments usage =
 (* This function is almost the same as [Arg.parse_expand], except
    that [Arg.parse_expand] could not be used because it does not take a
    reference for [arg_spec].*)
-let parse_arguments f msg =
+let parse_arguments argv f msg =
   try
-    let argv = ref Sys.argv in
+    let argv = ref argv in
     let current = ref (!Arg.current) in
```

Comment:
Should [current] be passed as an argument too? It's easy to forget to update Arg.current when calling main from the client of compiler-libs, possibly with a different argument array rather than Sys.argv.

Reply:
I've changed it to always start from 0. Since we're passing in argv, anyone whose wants to chop off some arguments can do so with Array.sub before here.

Reply:
Yes, makes sense. I think the code that calls Clflags.parse_arguments in makedepend.ml needs to be adjusted this way, as makedepend advances Arg.current and fiddles with Sys.argv before calling parse_arguments.
https://github.com/ocaml/ocaml/blob/6f3fc918d8ebc811d5a271b447d831ff1f040c65/driver/makedepend.ml#L646

Reply:
Nice catch! Thanks